File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/KronikHastaliklarFormViewModel.ts
//$DB94C753
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { KronikHastaliklarVeriSeti } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSSpirometri } from 'NebulaClient/Model/AtlasClientModel';
import { Type } from 'NebulaClient/ClassTransformer';

export class KronikHastaliklarFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.KronikHastaliklarFormViewModel, Core';
    @Type(() => KronikHastaliklarVeriSeti)
    public _KronikHastaliklarVeriSeti: KronikHastaliklarVeriSeti = new KronikHastaliklarVeriSeti();
    @Type(() => SKRSSpirometri)
    public SKRSSpirometris: Array<SKRSSpirometri> = new Array<SKRSSpirometri>();
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/KronikHastaliklarFormViewModel.cs
//$6C71B92C
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;

using Infrastructure.Helpers;
namespace Core.Controllers
{
    public partial class KronikHastaliklarVeriSetiServiceController
    {
        partial void PreScript_KronikHastaliklarForm(KronikHastaliklarFormViewModel viewModel, KronikHastaliklarVeriSeti kronikHastaliklarVeriSeti, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionID = viewModel.GetSelectedEpisodeActionID();
            if (selectedEpisodeActionID.HasValue)
            {
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionID.Value);
                viewModel._KronikHastaliklarVeriSeti.EpisodeAction = episodeAction;
                viewModel._KronikHastaliklarVeriSeti.SubEpisode = episodeAction.SubEpisode;
            }

            kronikHastaliklarVeriSeti.PaketeAitIslemZamani = DateTime.Now;
        }
    }
}

namespace Core.Models
{
    public partial class KronikHastaliklarFormViewModel:BaseViewModel, IENabizViewModel
    {
        public void AddENabizObjectViewModelToContext(TTObjectContext objectContext)
        {
            objectContext.AddObject(this._KronikHastaliklarVeriSeti);

            var kronikHastaliklar = (KronikHastaliklarVeriSeti)objectContext.GetLoadedObject(this._KronikHastaliklarVeriSeti.ObjectID);
            new SendToENabiz(objectContext, kronikHastaliklar.SubEpisode, kronikHastaliklar.ObjectID, kronikHastaliklar.ObjectDef.Name, "235", Common.RecTime());
        }
    }
}
